### ISSUE TITLE
[Bug]: llm completion exception breaks CodeActAgent

### ISSUE THREAD

**Issue by tobitege** [All-Hands-AI/OpenHands#3575]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

Any exception raised by `completion` within the CodeActAgent breaks the agent and is not recoverable, i.e. the page needs reloading to start a new session.

```        response = self.llm.completion(```
https://github.com/All-Hands-AI/OpenHands/blob/7589be671ecc0d226ad73f52e2fd89a1614bf4a4/agenthub/codeact_agent/codeact_agent.py#L175

### Current OpenHands version

```bash
0.9.0
```

### Installation and Configuration

```bash
any
```

### Operating System

any

### Logs, Errors, Screenshots, and Additional Context

Example exception log excerpt:
```py
CodeActAgent LEVEL 0 LOCAL STEP 12 GLOBAL STEP 12


Give Feedback / Get Help: https://github.com/BerriAI/litellm/issues/new
LiteLLM.Info: If you need to debug this error, use `litellm.set_verbose=True'.

Traceback (most recent call last):
  File "/home/tobias/.cache/pypoetry/virtualenvs/openhands-oz9p9M-K-py3.11/lib/python3.11/site-packages/litellm/llms/openai.py", line 1033, in completion
    raise e
  File "/home/tobias/.cache/pypoetry/virtualenvs/openhands-oz9p9M-K-py3.11/lib/python3.11/site-packages/litellm/llms/openai.py", line 966, in completion
    return convert_to_model_response_object(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tobias/.cache/pypoetry/virtualenvs/openhands-oz9p9M-K-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 5972, in convert_to_model_response_object
    raise raised_exception
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tobias/.cache/pypoetry/virtualenvs/openhands-oz9p9M-K-py3.11/lib/python3.11/site-packages/litellm/main.py", line 1923, in completion
    response = openai_chat_completions.completion(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tobias/.cache/pypoetry/virtualenvs/openhands-oz9p9M-K-py3.11/lib/python3.11/site-packages/litellm/llms/openai.py", line 1039, in completion
    raise OpenAIError(status_code=e.status_code, message=str(e))
litellm.llms.openai.OpenAIError
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/mnt/d/github/OpenDevin/openhands/controller/agent_controller.py", line 153, in _start_step_loop
    await self._step()
  File "/mnt/d/github/OpenDevin/openhands/controller/agent_controller.py", line 462, in _step
    action = self.agent.step(self.state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/d/github/OpenDevin/agenthub/codeact_agent/codeact_agent.py", line 175, in step
    response = self.llm.completion(
               ^^^^^^^^^^^^^^^^^^^^
 ```

**Comment by shubhamofbce** [comment]
@tobitege I will look into this today.

**Comment by tobitege** [comment]
Resolved by #3678 thanks to @shubhamofbce 🎉
